### PR TITLE
Region Params requirements

### PR DESCRIPTION
**Pull request by andymck**
Adds proto def for region params RPC

### CODE REVIEW

File: src/service/gateway.proto [hunk 1]
```diff
@@ -13,8 +14,25 @@ enum close_state {
   close_state_dispute = 3;
 }
 
+enum error_code {
+  error_generic = 0;
```
Comment (madninja) [inline src/service/gateway.proto:18]: why prefix it all with `error_` if it's already in the error_code enum?

File: src/service/gateway.proto [hunk 2]
```diff
@@ -13,8 +14,25 @@ enum close_state {
...
+  error_invalid_signature = 1;
+  error_not_asserted = 2;
+  error_unknown_region = 3;
+}
+
 /* general */
 
+message gateway_success_resp {
```
Comment (madninja) [inline src/service/gateway.proto:26]: not needed as part of this PR right? If so, what are they actually useful for? Generic "bytes" on success is quite vague?

File: src/service/gateway.proto [hunk 3]
```diff
@@ -13,8 +14,25 @@ enum close_state {
...
+  bytes resp = 1;
+  bytes details = 2;
+}
+
+message gateway_error_resp {
```
Comment (madninja) [inline src/service/gateway.proto:31]: ditto? Not needed in this PR?

File: src/service/gateway.proto (hunk 1 above)
Comment (andymck) [reply src/service/gateway.proto:18]: no hard reason other than the error prefix makes it easier to convey the msgs are a control set on the client side

File: src/service/gateway.proto (hunk 2 above)
Comment (andymck) [reply src/service/gateway.proto:26]: success resp is not required but the error resp is required and these two go together.

File: src/service/gateway.proto (hunk 3 above)
Comment (andymck) [reply src/service/gateway.proto:31]: see above

File: src/service/gateway.proto (hunk 2 above)
Comment (andymck) [reply src/service/gateway.proto:26]: success_resp is used for RPCs where there is no defined response payload to be returned.  For example when submitting a POC witness or receipt report.  In that example 'resp' has a value of ok....prob does make sense to do something else there than bytes

Comment (madninja) [reply src/service/gateway.proto:26]: how about just an empty success response?

Comment (andymck) [reply src/service/gateway.proto:26]: yeah prob as much as it needs.  will roll with that

**Pull request merged**